compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -885,12 +885,27 @@ int32_t TR_OSRLiveRangeAnalysis::perform()
             //
             _liveVars->empty();
             TR::Block *succ;
-            for (auto edge = block->getSuccessors().begin(); edge != block->getSuccessors().end(); ++edge)
+            auto edge = block->getSuccessors().begin();
+            for (; edge != block->getSuccessors().end(); ++edge)
                {
                succ = toBlock((*edge)->getTo());
                *_liveVars |= *liveLocals._blockAnalysisInfo[succ->getNumber()];
                }
 
+            // If the block contains an OSR point, it should have an exception edge
+            // to an OSR catch block
+            for (edge = block->getExceptionSuccessors().begin(); edge != block->getExceptionSuccessors().end(); ++edge)
+               {
+               if (toBlock((*edge)->getTo())->isOSRCatchBlock())
+                  break;
+               }
+            if (edge == block->getExceptionSuccessors().end())
+               {
+               if (comp()->getOption(TR_TraceOSR))
+                  traceMsg(comp(), "skipping block_%d without OSR points", block->getNumber());
+               break;
+               }
+
             if (comp()->getOption(TR_TraceOSR))
```

Comment:
Is'nt there some maintainLiveness logic later on that is'nt predicated on an OSR point being present in the block ? If so, why is it okay to skip that code ?

Reply:
`maintainLiveness` is only concerned with retaining liveness between treetops in a block. When a new block is seen, `_liveVars` is reset according to the liveness results for the block's successors. Therefore, blocks can be processed out of order and skipped.